### DISCUSSION TITLE
Resetting password doesn't log causer_type.

### DISCUSSION THREAD

**Discussion by airliajsmith** [spatie/laravel-activitylog#812] [Q&A] [answered]
When a user resets their password, the activity is logged, but the causer_type is null. I haven't had this issue with any other logging activity. Any idea as to why?

**Comment by Gummibeer** [comment]
Without the code you are using to reset the password and log the activity we can't help you.
Do you use model change logging or a custom activity log?

**Comment by airliajsmith** [comment]
Sorry, I thought about that after I posted.

I use Laravel's built-in system for changing the password. I added this to the User model, but I had the same results before I added it:

    use Notifiable, HasRoles, SoftDeletes, LogsActivity, HasSlug, HasFactory;

    protected static $logName = 'user';
    protected static $logFillable = true;
    protected static $ignoreChangedAttributes = ['created_at', 'updated_at', 'password'];
    protected static $submitEmptyLogs = false;
    protected static $logOnlyDirty = true;

    public function tapActivity(Activity $activity)
    {
        if ($properties = $activity->properties) {
            if ($properties->has('attributes')) {
                $attributes = $properties->get('attributes');
                if (isset($attributes['password'])) {
                    $attributes['password'] = '<secret>';
                }
                $properties->put('attributes', $attributes);
            }
            if ($properties->has('old')) {
                $old = $properties->get('old');
                if (isset($old['password'])) {
                    $old['password'] = '<secret>';
                }
                $properties->put('old', $old);
            }
            $activity->properties = $properties;
        }
    }

**Comment by Gummibeer** [comment, marked as answer]
I have read your initial question wrong. 🤦‍♂️
You are talking about the `causer` - during password forget/reset no user is authenticated (otherwise he would use password change). That's why there's no `causer` during password reset.

**Comment by airliajsmith** [comment]
Is there a way to set the causer in this instance? I want to set it to App\Models\User.

**Reply by Gummibeer** [reply]
You have two ways - you can use the `tapActivity()` method to set it or use a custom activity with the corresponding builder methods.

**Reply by airliajsmith** [reply]
Can I add to the code below? If so, can you provide the code?

    public function tapActivity(Activity $activity)
    {
        if ($properties = $activity->properties) {
            if ($properties->has('attributes')) {
                $attributes = $properties->get('attributes');
                if (isset($attributes['password'])) {
                    $attributes['password'] = '<secret>';
                }
                $properties->put('attributes', $attributes);
            }
            if ($properties->has('old')) {
                $old = $properties->get('old');
                if (isset($old['password'])) {
                    $old['password'] = '<secret>';
                }
                $properties->put('old', $old);
            }
            $activity->properties = $properties;
        }
    }
